tests/test_data.py

```diff
@@ -134,5 +134,30 @@ def test_average_by(self):
         self.assertEqual(descriptor[-1],5)
         assert(np.all(self.test_data.measurements[-1]==avg[-1]))
         
+
+class testSave(unittest.TestCase):
+    def test_save_load(self):
+        measurements = np.zeros((10,5))
+        des = {'session':0,'subj':0}
+        obs_des = {'conds':np.array([0,0,1,1,2,2,2,3,4,5])}
+        chn_des = {'rois':np.array(['V1','V1','IT','IT','V4'])}
+        data = rsd.Dataset(measurements=measurements,
+                           descriptors=des,
+                           obs_descriptors=obs_des,
+                           channel_descriptors=chn_des
+                           )
+        try:
+            data.save('test')
+            data_loaded = rsd.load_dataset('test')
+        finally:
+            import os
+            os.remove('test.npy')
+            os.remove('test_desc.pkl')
```

Comment:
I predict that this is gonna cause trouble.. It's good that it's in a finally loop, but that won't help against `ctrl-c`. Another way is to store them in an in-memory file object like `io.BytesIO`. If the filesystem interaction bit is a separate class you could have a MockFilesystem or something, or alternatively you could `@patch('my_module.open')`

Reply:
Maybe don't worry about this right now but keep it in mind if there's gonna be more file I/O

Reply:
I hope that this pull request does most of the file i/o. The in memory idea sounds good to me. That would not easily with multiple files I guess or at least not as an actual test. I will come back to this once I tried the hdf5 idea. If this allows me to save stuff into one file, I should be able to make this test work in memory only. Thanks!

Reply:
now switched also to saving into `io.BytesIO` objects